### DISCUSSION TITLE
how do i delete a remote user and all their posts?

### DISCUSSION THREAD

**Discussion by ghost** [misskey-dev/misskey#8539] [Q&A] [answered]
i have the option to suspend a user, but how do i delete a remote user and all of their notes?

**Comment by Johann150** [comment, marked as answer]
If you want to delete a user, you will have to manually delete them in the database. Note however that they might federate to your instance again in the future in which case Misskey will treat the user as any other newly recognized user, and will not remember that you suspended the user before.

Note also that any content by a suspended user can only be viewed on your instance by administrators or moderators, so just suspending a user may be more desirable.

If you delete the user's record from the `user` table, their notes will be automatically deleted by PostgreSQL due to a foreign key constraint.

**Reply by ghost** [reply]
Thank you very much for the answer. I do not mind if the user gets detected again, I just want to delete their past notes, so this option suits me best.

**Reply by Johann150** [reply]
For completeness: You can delete a user with this SQL query: 
```sql
DELETE FROM "user" WHERE id = '<id>';
```
Replace `<id>` with the ID of the user (leave the single quotes). You can find the ID of a user by going to their profile page, click the three dots next to the follow button and select the "About" menu entry.
You should be absolutely sure as this is not a reversible operation. You should make a backup.

**Comment by ThatOneCalculator** [comment]
Follow up: Deleting users, both local and remote, is now a feature in the 12.112 beta. Just go to the user's profile, 3 dots > about, Moderation > Delete account. 🎉

**Reply by spirillen** [reply]
But we still need the feature to delete all the notes created by any users, which we have either `Suspended` (preferable) or deleted. IE I have a clickbait account from @pubeurope.com with about 4000 clickbait notes, the only way I see it realistic to delete these at the moment , is to go into the database and do this.

```sql
DELETE FROM public.note
    WHERE "userId" LIKE '9k7rfz2mbz';
```

But then what about any of the other fields that can/will/is associated to other tables??


```
"replyId", "renoteId", "text", "name", cw, "userId", "localOnly", "renoteCount", "repliesCount", reactions, visibility, uri, "fileIds", "attachedFileTypes", "visibleUserIds", mentions, "mentionedRemoteUsers", emojis, tags, "hasPoll", "userHost", "replyUserId", "replyUserHost", "renoteUserId", "renoteUserHost", url, "channelId", "threadId", "reactionAcceptance", "clippedCount", "reactionAndUserPairCache"
```
